### PR TITLE
Move docs to mkdocs base

### LINKED ISSUE #132: Improve NoitaMP README.md and docs with MkDocs!
**Is your feature request related to a problem? Please describe.**
Not problem at all, just something fancy!

**Describe the solution you'd like**
Would be awesome to use [MkDocs](https://github.com/squidfunk/mkdocs-material) instead of overloaded [README.md](README.md) for NoitaMP.
README.md then simply has a quick and clear description with a link to the docs.

### PR DESCRIPTION
**Pull request by Furrior**
Fixes #132

Moved most of the stuff to the mkdocs. Fixed a lot of .md linter issues.

[Priview new docs](https://furrior.github.io/NoitaMP/)

### CODE REVIEW

**Comment by Ismoh** [comment]
@Furrior ! Thanks a ton for contributing and investigating time!

I really appreciate your PR. 
Unfortunatelly you need to pull/update/rebase master on your changes again.
Version mismatch and other stuff. I've updated master with develop some seconds ago. My bad!

FYI:
> Before contributing
> Please get in contact with me or the other collaborators on [Discord](https://discord.gg/DhMurdcw4k), 
> before you start working on a feature or a bug fix. This is to avoid duplicate work.

[CONTRIBUTING.md](https://github.com/Ismoh/NoitaMP/blob/master/CONTRIBUTING.md#before-contributing)

**Comment by Furrior** [comment]
> @Furrior ! Thanks a ton for contributing and investigating time!
> 
> I really appreciate your PR. Unfortunatelly you need to pull/update/rebase master on your changes again. Version mismatch and other stuff. I've updated master with develop some seconds ago. My bad!
> 
> FYI:
> 
> > Before contributing
> > Please get in contact with me or the other collaborators on [Discord](https://discord.gg/DhMurdcw4k),
> > before you start working on a feature or a bug fix. This is to avoid duplicate work.
> 
> [CONTRIBUTING.md](https://github.com/Ismoh/NoitaMP/blob/master/CONTRIBUTING.md#before-contributing)

Updating isnt a big deal anyway, so no problem. I did search through discord just with discord search and found no mentions of this issue except creating it via some bot. Probably still needed to ask, but yeah...

**Comment by Furrior** [comment]
So I deployed it on my fork so you can check how it looks like: https://furrior.github.io/NoitaMP/
Need to test gh workflows and then it will be ready I guess.

**Comment by Ismoh** [comment]
@Furrior I'll do review soon and let you when done.

**Comment by Furrior** [comment]
Tested that generating changelog still works as intended after changing the path
https://github.com/Furrior/NoitaMP/commit/326c6adeb7236b3f8ff09922b29b797839f18f14

File: README.md [hunk 1]
```diff
@@ -61,73 +58,6 @@ Let's see, if I can do so?! I am new to Lua and modding, but someone said to me:
...
-- @Coxas/Thighs
-- [@dextercd](https://github.com/dextercd)
-- [@BlueAmulet](https://github.com/BlueAmulet)
-- [@Shaw](https://github.com/ShawSumma)
-- [@Ramiels](https://github.com/Ramiels)
-- [@bruham](https://steamcommunity.com/id/bruham/myworkshopfiles/?appid=881100)
-- [@ofoxsmith](https://github.com/ofoxsmith)
-- [@Myzumi](https://github.com/Myzumi)
-
-I also want to say thank you to the sponsors of this project on
```
Comment (Ismoh) [inline README.md:88]: A reference (link) on README.md to this section (sponsor links) in mkdocs would be good. Can you add it?

File: docs/contributing/CONTRIBUTING.md [hunk 2]
```diff
@@ -1,79 +1,90 @@
...
 ### Lua context
+
 There are different Lua contexts in Noita, but the most important one is the `init.lua` context.
 
 ### Contexts in Noita
 
 Let me try to explain it with an example:
-![NoitaMP](miscs/contributing-md/lua-contexts.png)
-In the diagram above you can see the default Noita modding structure.\
+![NoitaMP](/resources/img/lua-contexts.png)
```
Comment (Ismoh) [inline docs/contributing/CONTRIBUTING.md:46]: Change resources to `res` please.

File: docs/contributing/CONTRIBUTING.md [hunk 3]
```diff
@@ -1,79 +1,90 @@
...
 
 #### GlobalsSetValue and GlobalsGetValue
+
 Then we would set a global "foo" variable in the `LuaComponent` context _(see 1 and 2 in the diagram below)_
-and afterwards we would use the `init.lua` context to fetch the value of the global "foo" variable _(see 3 and 4 in the diagram below)_.\
+and afterwards we would use the `init.lua` context to fetch the value of the global "foo" variable _(see 3 and 4 in the diagram below)_.
 In addition, please note that `GlobalsSetValue` and `GlobalsGetValue` has nothing to do with Lua globals `_G`.
 See diagram below for a better understanding:
-![NoitaMP](miscs/contributing-md/lua-contexts-workaround.png)
+![NoitaMP](/resorces/img/lua-contexts-workaround.png)
```
Comment (Ismoh) [inline docs/contributing/CONTRIBUTING.md:81]: `res`

File: docs/index.md [hunk 4]
```diff
@@ -2,60 +2,8 @@
...
-| Toggle multiplayer menu              | Not possible atm                                                                                       |
-| Debug settings                       |                                                                                                        |
-| Toggle debug (in game)               | Enable nuid debugger in a run. Nothing you can do with :)                                              |
-| Log level                            | Will be interesting when something isnt working, but for higher fps and performance, keep it on Error! |
- 
-## Support
-
-If you need help or anything looks like to be broken, search the [issues](https://github.com/Ismoh/NoitaMP/issues?q=is%3Aissue) to find a similar problem or [create](https://github.com/Ismoh/NoitaMP/issues/new/choose) a new issue, when there really is something broken! Please do me a favour and first check the old issues. I appreciate it!
+If you want to get notifications, [join](https://discord.gg/DhMurdcw4k) my discord server. There is a MEE6 role assigment for NoitaMP in #noita-modding!
```
Comment (Ismoh) [inline docs/index.md:9]:
```suggestion
If you want to get notifications, [join](https://discord.gg/DhMurdcw4k) my discord server. There is a carl.gg role assigment for NoitaMP in #noita-modding!
```

File: LICENSE.md
Comment (Ismoh) [inline LICENSE.md:1]: (README.md,) LICENSE.md, SUPPORT.md and CODE_OF_CONDUCT.md must be on repo root dir, otherwise this won't work anymore:

image

Add a command to `deploy_mkdocs.yml` to copy those files?! I don't want to maintain copies.

File: SUPPORT.md
Comment (Ismoh) [inline SUPPORT.md:1]: (README.md,) LICENSE.md, SUPPORT.md and CODE_OF_CONDUCT.md must be on repo root dir, otherwise this won't work anymore:

image

Add a command to `deploy_mkdocs.yml` to copy those files?! I don't want to maintain copies.

File: .github/workflows/deploy_mkdocs.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Configure Git Credentials
+        run: |
+          git config user.name github-actions[bot]
+          git config user.email 41898282+github-actions[bot]@users.noreply.github.com
+      - uses: actions/setup-python@v4
```
Comment (Ismoh) [inline .github/workflows/deploy_mkdocs.yml:19]: Add a step to copy to `/docs`, because:

(README.md,) LICENSE.md, SUPPORT.md and CODE_OF_CONDUCT.md must be on repo root dir, otherwise this won't work anymore:

image

Add a command to `deploy_mkdocs.yml` to copy those files?! I don't want to maintain copies.

File: .github/workflows/version-and-changelog-update.yml [hunk 6]
```diff
@@ -199,7 +199,7 @@ jobs:
         with:
           token: ${{ secrets.GITHUB_TOKEN }}
           dateFormat: "%d.%m.%Y"
-          output: "CHANGELOG.md"
+          output: "docs/CHANGELOG.md"
```
[the remark this answers is not in the document]
Comment (Ismoh) [reply .github/workflows/version-and-changelog-update.yml:202]: Would also keep this on root repo dir and do a copy in deploy_mkdocs.yml. Just for the github nerds!

File: docs/contributing/CONTRIBUTING.md
Comment (Ismoh) [inline docs/contributing/CONTRIBUTING.md:1]: CONTRIBUTING.md is also a github default. Keep it on repo root and do a copy in `deploy_mkdocs.yml`

(README.md,) LICENSE.md, SUPPORT.md and CODE_OF_CONDUCT.md must be on repo root dir, otherwise this won't work anymore:

image

Add a command to `deploy_mkdocs.yml` to copy those files?! I don't want to maintain copies.

File: docs/index.md [hunk 7]
```diff
@@ -2,60 +2,8 @@
 
 NoitaMP is a mod for [Noita](https://noitagame.com/) to be able to share your feelings while playing Noita with your friends. It's a multiplayer mod :rage1: View it on [GitHub](https://github.com/Ismoh/NoitaMP).
 
-## Installation
+## TL;DR
```
Comment (Ismoh) [inline docs/index.md:5]:
```suggestion
## TL;DR / Installation
```

File: mkdocs.yml [hunk 8]
```diff
@@ -0,0 +1,24 @@
+site_name: NoitaMP Documentation
+
+theme:
+  name: material
+  logo: 'resources/img/NoitaMP.png'
```
Comment (Ismoh) [inline mkdocs.yml:5]: Images on your fork deployment are broken. Did you recognised?

**Review by Ismoh — CHANGES_REQUESTED** [review]
Thanks again for your help!

I am happy with this PR, besides small changes. I added plenty of comments. Please have a look. 🥇

**Comment by Furrior** [comment]
Not sure about the place for credits link in readme and that understood you correctly with custom field in funding

**Comment by Ismoh** [comment]
> Not sure about the place for credits link in readme and that understood you correctly with custom field in funding

https://docs.github.com/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/displaying-a-sponsor-button-in-your-repository#about-funding-files

You're right! Thanks!
I'll take a look soon! 💯

File: docs/index.md [hunk 9]
```diff
@@ -2,60 +2,8 @@
...
-| Toggle multiplayer menu              | Not possible atm                                                                                       |
-| Debug settings                       |                                                                                                        |
-| Toggle debug (in game)               | Enable nuid debugger in a run. Nothing you can do with :)                                              |
-| Log level                            | Will be interesting when something isnt working, but for higher fps and performance, keep it on Error! |
- 
-## Support
-
-If you need help or anything looks like to be broken, search the [issues](https://github.com/Ismoh/NoitaMP/issues?q=is%3Aissue) to find a similar problem or [create](https://github.com/Ismoh/NoitaMP/issues/new/choose) a new issue, when there really is something broken! Please do me a favour and first check the old issues. I appreciate it!
+If you want to get notifications, [join](https://discord.gg/DhMurdcw4k) my discord server. There is a carl.gg role assigment for NoitaMP in #noita-modding!
```
Comment (Ismoh) [inline docs/index.md:9]:
```suggestion
If you want to get notifications, [join](https://discord.gg/DhMurdcw4k) my discord server. There is a carl.gg role assigment for NoitaMP in #noitamp-chat!
```

**Review by Ismoh — APPROVED** [review]
Thanks for work, I really appreciate it a lot!
PR approved! 💯

**Comment by Furrior** [comment]
изображение
Skill issue happened
